homeassistant/components/media_player/hdmi_cec.py

```diff
@@ -26,9 +26,17 @@ def setup_platform(hass, config, add_entities, discovery_info=None):
     """Find and return HDMI devices as +switches."""
     if ATTR_NEW in discovery_info:
         _LOGGER.info("Setting up HDMI devices %s", discovery_info[ATTR_NEW])
-        add_entities(CecPlayerDevice(hass, hass.data.get(device),
-                                     hass.data.get(device).logical_address) for
-                     device in discovery_info[ATTR_NEW])
+        for device in discovery_info[ATTR_NEW]:
+            hdmi_device = hass.data.get(device)
+            entity = CecPlayerDevice(
+                hass,
+                hdmi_device,
+                hass.data.get(device).logical_address,
+            )
+            add_entities([entity])
```

Comment:
This is not good. Call `add_entities` with all entities at once and just pass `True` as second param to update the entities in sequence.

Reply:
https://github.com/home-assistant/home-assistant/blob/43676fcaf402994f18735bbd71a62fca54567594/homeassistant/helpers/entity_platform.py#L192 tells that `update_before_add` should only be called inside tests.